weblab/datasets/views.py

```diff
@@ -0,0 +1,143 @@
+import json
+import mimetypes
+import os.path
+import shutil
+import subprocess
+from itertools import groupby
+from tempfile import NamedTemporaryFile
+
+import requests
+from braces.views import UserFormKwargsMixin
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth.mixins import (
+    LoginRequiredMixin,
+    PermissionRequiredMixin,
+    UserPassesTestMixin,
+)
+from django.core.urlresolvers import reverse
+from django.http import (
+    Http404,
+    HttpResponse,
+    HttpResponseBadRequest,
+    HttpResponseRedirect,
+    JsonResponse,
+)
+from django.core.exceptions import PermissionDenied
+from django.db.models import F, Q
+from django.utils.decorators import method_decorator
+from django.utils.text import get_valid_filename
+from django.views import View
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import TemplateView
+from django.views.generic.base import RedirectView
+from django.views.generic.detail import DetailView, SingleObjectMixin
+from django.views.generic.edit import CreateView, DeleteView, FormMixin
+from django.views.generic.list import ListView
+from git import BadName, GitCommandError
+from guardian.shortcuts import get_objects_for_user
+
+from core.filetypes import get_file_type
+from core.visibility import (
+    Visibility, VisibilityMixin
+)
+from experiments.models import Experiment, PlannedExperiment
+from repocache.exceptions import RepoCacheMiss
+from repocache.models import CachedEntityVersion
+
+from .models import ExperimentalDataset
+
+from .forms import (
+    FileUploadForm,
+)
+
+class ExperimentalDatasetCreateView(
+    LoginRequiredMixin, PermissionRequiredMixin,
+    UserFormKwargsMixin, CreateView
+):
+    """
+    Create new ExperimentalDataset
+    """
+    template_name = 'datasets/dataset_form.html'
+
+    @property
+    def permission_required(self):
+        datasets.create_dataset_experiment
+
+    @property
+    def form_class(self):
+        return FormMixin
+
+    def get_success_url(self):
+        return reverse('datasets:newversion',
+                       args=[self.object.pk])
+
+
+class ExperimentalDatasetListView(LoginRequiredMixin, ListView):
+    """
+    List all user's datasets
+    """
+    template_name = 'datasets/dataset_list.html'
+
+    def get_queryset(self):
+        return self.model.objects.filter(author=self.request.user)
+
+
+# class ExperimentalDatasetView(VisibilityMixin, SingleObjectMixin, RedirectView):
+#     """
+#     View an ExperimentalDataset
+#
+#     All this does is redirect to the latest version of the entity, if it exists.
```

Comment:
We don't have versions for datasets, so this should *be* the equivalent of the latest version view.